src/consensus.rs

```diff
@@ -877,8 +877,54 @@ struct RaftMessageSender {
 
 impl RaftMessageSender {
     pub async fn exec(mut self) {
-        while let Some(message) = self.messages.recv().await {
-            self.send(&message).await;
+        let mut heartbeat = None;
+
+        loop {
```

Comment:
I'd like a description of the state machine in one place here better. It would make the logic in the loop a lot smaller and likely easier to go through and understand. What do you think?

Reply:
The only way to know is to try it and then see which one we like more/makes more sense, I think.

Reply:
The code, suddenly, become much more trivial. ¯\\\_(ツ)\_/¯